packages/react-native-drawer-layout/src/views/GestureHandler.tsx

```diff
@@ -1,22 +1,37 @@
 import * as React from 'react';
 import { View } from 'react-native';
-import type {
-  PanGestureHandlerProperties,
-  TapGestureHandlerProperties,
-} from 'react-native-gesture-handler';
+// I think it needs to be imported from compiled types
+import type { ComposedGesture } from 'react-native-gesture-handler';
+import type { UserSelect } from 'react-native-gesture-handler/lib/typescript/handlers/gestureHandlerCommon';
+import type { GestureType } from 'react-native-gesture-handler/lib/typescript/handlers/gestures/gesture';
+export type { PanGesture } from 'react-native-gesture-handler/lib/typescript/handlers/gestures/panGesture';
 
 const Dummy: any = ({ children }: { children: React.ReactNode }) => (
   <>{children}</>
 );
 
-export const PanGestureHandler =
-  Dummy as React.ComponentType<PanGestureHandlerProperties>;
+export type GestureDetectorProps = {
+  gesture: GestureType | ComposedGesture;
+  userSelect?: UserSelect;
+  children?: React.ReactNode;
+};
 
-export const TapGestureHandler =
-  Dummy as React.ComponentType<TapGestureHandlerProperties>;
+export type MaybeGestureDetectorProps = {
+  gesture: GestureType | ComposedGesture | undefined;
+  userSelect?: UserSelect;
+  children?: React.ReactNode;
+};
+
+export const GestureDetector = Dummy as (
+  props: GestureDetectorProps
+) => React.ReactNode;
 
 export const GestureHandlerRootView = View;
 
+export const Gesture = undefined as
```

Comment:
should be fine without `as`

```suggestion
export const Gesture: typeof import(https://github.com/software-mansion/react-native-gesture-handler).Gesture | undefined = undefined.
```

Reply:
probably not url?